pkg/kritis/cryptolib/verifier_test.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cryptolib
+
+import (
+	"testing"
+
+	"github.com/pkg/errors"
+)
+
+func TestVerifyAttestation(t *testing.T) {
+	att := &Attestation{
+		PublicKeyID:       "key-id",
+		Signature:         []byte("signature"),
+		SerializedPayload: []byte("payload"),
+	}
+	matchingKey := NewPublicKey(Pkix, []byte("key-data"), "key-id")
+	otherMatchingKey := NewPublicKey(Pkix, []byte("key-data-other"), "key-id")
+	nonmatchingKey := NewPublicKey(Pkix, []byte("key-data-other"), "key-id-other")
+
+	tcs := []struct {
+		name         string
+		att          *Attestation
+		publicKeys   []PublicKey
+		pkixVerifier pkixVerifier
+		expectedErr  bool
+	}{
+		{
+			name:         "single key match",
+			att:          att,
+			publicKeys:   []PublicKey{matchingKey},
+			pkixVerifier: testVerifierPass{},
+			expectedErr:  false,
+		},
+		{
+			name:         "matching and nonmatching keys",
+			att:          att,
+			publicKeys:   []PublicKey{nonmatchingKey, matchingKey},
+			pkixVerifier: testVerifierPass{},
+			expectedErr:  false,
+		},
+		{
+			// This is a possibility with user-provided IDs
+			name:         "different keys with same ID",
+			att:          att,
+			publicKeys:   []PublicKey{matchingKey, otherMatchingKey},
+			pkixVerifier: testVerifierPass{},
+			expectedErr:  false,
+		},
+		{
+			name:         "key not found",
+			att:          att,
+			publicKeys:   []PublicKey{nonmatchingKey},
+			pkixVerifier: testVerifierPass{},
+			expectedErr:  true,
+		},
+		{
+			name:         "error in verification",
+			att:          att,
+			publicKeys:   []PublicKey{matchingKey},
+			pkixVerifier: testVerifierFail{},
+			expectedErr:  true,
+		},
+	}
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			v, err := NewVerifier("test-image-digest", tc.publicKeys)
```

Comment:
Instead of calling NewVerifier and casting, manually construct the internal verifier struct. tc.publicKeys can be a map or you can factor out an internal slice to map function that NewVerifier and the tests both call

Reply:
Done.